.github/workflows/ci.yml

```diff
@@ -16,8 +16,8 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
-permissions:
-  contents: read
+# Declare default permissions as read only
+permissions: read-all

```

Comment:
why `read-all` and not only `read`? My understanding is that `read` is even more restrictive.

Reply:
you're absolutely right. I had assumed that the other scopes would retain default access, but it turns out they have no access unless explicitly specified. I've reverted the changes and updated the configuration as you suggested [set least privilege](https://github.com/Open-CMSIS-Pack/vscode-cmsis-debugger/pull/236/commits/82e8de1017f6e0039b0b095f160931ca223bf362)

Reply:
Hm....looks like something's now too restrictive. Can't see any successful CI workflows since the latest changes.

Reply:
Seems to be working again after last push.